hdf/src/atom.c

```diff
@@ -53,11 +53,63 @@ MODIFICATION HISTORY
    1/7/96  - Finished coding prototype
 */
 
-#define ATOM_MASTER
 #include "hdf.h"
 #include "atom.h"
+
 #include <assert.h>
 
+/* # of bits to use for Group ID in each atom (change if MAXGROUP>16) */
+#define GROUP_BITS 4
+#define GROUP_MASK 0x0F
+
+/* # of bits to use for the Atom index in each atom (change if MAXGROUP>16) */
+#define ATOM_BITS 28
+#define ATOM_MASK 0x0FFFFFFF
+
+/* # of previous atoms cached, change inline caching macros (HAatom_object & HAIswap_cache) if this changes */
+#define ATOM_CACHE_SIZE 4
+
+/* Map an atom to a Group number */
+#define ATOM_TO_GROUP(a) ((group_t)((((atom_t)(a)) >> ((sizeof(atom_t) * 8) - GROUP_BITS)) & GROUP_MASK))
```

Comment:
Maybe for a future PR: Consider converting local `#define` macros to `static inline` functions. I'm pretty sure `inline` came in with C99.